core/src/main/java/jenkins/security/s2m/DefaultFilePathFilter.java

```diff
@@ -59,6 +64,17 @@ protected boolean op(String op, File f) throws SecurityException {
                     LOGGER.log(Level.FINE, "agent allowed to {0} {1}", new Object[] {op, f});
                     return true;
                 } else {
+                    try {
+                        Field current = Request.class.getDeclaredField("CURRENT");
+                        current.setAccessible(true);
+                        Field createdAt = Command.class.getDeclaredField("createdAt");
+                        createdAt.setAccessible(true);
+                        Throwable trace = (Throwable) createdAt.get(((ThreadLocal) current.get(null)).get());
+                        ExtensionList.lookupSingleton(SlaveToMasterFileCallableUsage.class).recordTrace(trace);
+                        LOGGER.log(Level.WARNING, "Permitting agent-to-controller " + op + " on " + f, trace);
```

Comment:
```suggestion
                        LOGGER.log(Level.WARNING, "Permitting agent-to-controller '" + op + "' on '" + f + "'. This is deprecated and will soon be rejected. Learn more: https://www.jenkins.io/redirect/permitted-agent-to-controller-file-access", trace);
```

Give admins a chance to understand why this is a warning and why they should care.

Reply:
This is a 404.

Reply:
Will be a new page once this is merged.

Reply:
It doesn't, but we can easily create it. Doesn't make a lot of sense to create it before this change gets accepted (of course, we can prepare the corresponding PR). Could be as simple as redirecting to this PR, or my JEP PR, or the JEP once merged, to provide context.

#5885 links to `https://www.jenkins.io/redirect/AdminWhitelistRule` which doesn't exist yet either 🤷